File: servers/robot-controller-backend/gpio_mock.py
class GPIO:
    BCM = "BCM"
    OUT = "OUT"
    HIGH = True
    LOW = False

    @staticmethod
    def setmode(mode):
        print(f"GPIO setmode({mode})")

    @staticmethod
    def setup(pin, mode):
        print(f"GPIO setup({pin}, {mode})")

    @staticmethod
    def output(pin, state):
        print(f"GPIO output({pin}, {state})")

    @staticmethod
    def cleanup():
        print("GPIO cleanup()")

# Simulating the GPIO actions
def main():
    import sys
    pin = int(sys.argv[1])
    state = sys.argv[2].upper()

    GPIO.setmode(GPIO.BCM)
    GPIO.setup(pin, GPIO.OUT)

    if state == "HIGH":
        GPIO.output(pin, GPIO.HIGH)
    else:
        GPIO.output(pin, GPIO.LOW)

    GPIO.cleanup()

if __name__ == "__main__":
    main()

File: servers/robot-controller-backend/gpio_mock.go
package main

import (
    "github.com/stianeikeland/go-rpio/v4"
)

// MockGPIO implements the GPIO interface for testing purposes
type MockGPIO struct{}

func (m MockGPIO) Open() error {
    return nil
}

func (m MockGPIO) Close() error {
    return nil
}

func (m MockGPIO) Pin(pin int) GPIOPin {
    return MockGPIOPin{}
}

type MockGPIOPin struct{}

func (p MockGPIOPin) Input() {}

func (p MockGPIOPin) Output() {}

func (p MockGPIOPin) Read() rpio.State {
    return Low // Mock read to return Low state
}

func (p MockGPIOPin) High() {}

func (p MockGPIOPin) Low() {}
